source/agora/node/Ledger.d

```diff
@@ -105,19 +105,20 @@ public class Ledger
             }
         }
 
-        // restore validator set from lastest blocks
-        if (this.last_block.header.height > 0)
+        // max: there may be less than 1008 blocks stored in the ledger
+        const min_height = max(long(0), long(this.last_block.header.height)
+            - Enrollment.ValidatorCycle);
```

Comment:
Not a big fan of the `long(...) - X` thing, I'm more a `x >= y ? 0 : x - y` kind of person, but to each their own.

Reply:
> I'm more a `x >= y ? 0 : x - y` kind of person

Sure, but you wrote the wrong comparison there.

Reply:
I never said I was the right kind of person tho

Reply:
:P